gke-deploy/cmd/apply/apply.go

```diff
@@ -58,14 +58,14 @@ func NewApplyCommand() *cobra.Command {
 		SilenceUsage: true,
 	}
 
-	cmd.Flags().StringVarP(&options.filename, "filename", "f", "", "Configuration file or directory of configuration files to use to create Kubernetes objects (file or files in directory must end with \".yml\" or \".yaml\").")
+	cmd.Flags().StringVarP(&options.filename, "filename", "f", "", "Configuration file or directory of configuration files to use to create Kubernetes objects (file or files in directory must end with \".yml\" or \".yaml\"). The configuration file(s) will be download to a temporary folder first if the file or directory is a GCS URL, e.g. gs://bucket/directory/*.")
 	cmd.Flags().StringVarP(&options.clusterLocation, "location", "l", "", "Region/zone of GKE cluster to deploy to.")
 	cmd.Flags().StringVarP(&options.clusterName, "cluster", "c", "", "Name of GKE cluster to deploy to.")
 	cmd.Flags().StringVarP(&options.clusterProject, "project", "p", "", "Project of GKE cluster to deploy to. If this field is not provided, the current set GCP project is used.")
 	cmd.Flags().StringVarP(&options.namespace, "namespace", "n", "", "Namespace of GKE cluster to deploy to. If omitted, the namespace(s) specified in each Kubernetes configuration file is used.")
 	cmd.Flags().BoolVarP(&options.verbose, "verbose", "V", false, "Prints underlying commands being called to stdout.")
 	cmd.Flags().DurationVarP(&options.waitTimeout, "timeout", "t", 5*time.Minute, "Timeout limit for waiting for Kubernetes objects to finish applying.")
-	cmd.Flags().BoolVarP(&options.recursive, "recursive", "R", false, "Recursively search through the configuration directory for all yaml files.")
+	cmd.Flags().BoolVarP(&options.recursive, "recursive", "R", false, "Recursively search through the configuration directory or the GCS bucket(if \"filename\" is a GCS URL) for all YAML files. ")
```

Comment:
change

if \"filename\"

to

if --filename

Reply:
Recursively search through the configuration directory passed by --filename for all YAML files.

Reply:
I would...
s/configuration directory/configuration file or files/

That way it includes config downloaded from GCS, but if you just say "configuration directory" it sounds (to me) like it's assuming local file or dir.

Reply:
Agreed that that can "directory" can be misleading. I think "file or files" is also confusing because you can't recursively search through one file.

Maybe:

`Recursively search through the provided path in --filename for all YAML files.`

Reply:
Copy-pasted :)